contracts/FleekERC721.sol

```diff
@@ -157,24 +150,7 @@ contract FleekERC721 is ERC721, FleekAccessControl {
         address owner = ownerOf(tokenId);
         App storage app = _apps[tokenId];
 
-        // prettier-ignore
-        bytes memory dataURI = abi.encodePacked(
-            '{',
-                '"name":"', app.name, '",',
-                '"description":"', app.description, '",',
-                '"owner":"', Strings.toHexString(uint160(owner), 20), '",',
-                '"external_url":"', app.externalURL, '",',
-                '"image":"', app.image, '",',
-                '"attributes": [',
-                    '{"trait_type": "ENS", "value":"', app.ENS,'"},',
-                    '{"trait_type": "Commit Hash", "value":"', app.builds[app.currentBuild].commitHash,'"},',
-                    '{"trait_type": "Repository", "value":"', app.builds[app.currentBuild].gitRepository,'"},',
-                    '{"trait_type": "Version", "value":"', app.currentBuild.toString(),'"}',
-                ']',
-            '}'
-        );
-
-        return string(abi.encodePacked(_baseURI(), Base64.encode((dataURI))));
+        return string(abi.encodePacked(_baseURI(), app.toString(owner).toBase64()));
```

Comment:
Everything looks good to me. Maybe if this is the only spot we do app.toString() we could just have the library return this encodePacked already